### PR TITLE
Add: Include basic .editorconfig file to help enforce tab style

### PR DESCRIPTION
**Pull request by orudge**
As per http://editorconfig.org/, add a .editorconfig file to help coax editors into doing the right thing by default. :)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
+insert_final_newline = true
```
Comment (TrueBrain) [inline .editorconfig:4]: is eol also a good idea here?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+trim_trailing_whitespace = true
+
+[*.{c,cpp,h,hpp}]
```
Comment (TrueBrain) [inline .editorconfig:7]: how about charsets for these?

File: .editorconfig (hunk 1 above)
Comment (glx22) [reply .editorconfig:4]: the doc says to not set eol if we want native line endings

File: .editorconfig (hunk 2 above)
Comment (orudge) [reply .editorconfig:7]: I can add charset = utf-8 :)

**Comment by frosch123** [comment]
Wouldn't the charset go into the [*] section?
All our files should be UTF-8.

**Comment by orudge** [comment]
@frosch123 Generally, I would say so, yes. I didn't want to be too prescriptive in this file though in case, for any reason, we had some files that it caused problems for. (Windows .rc files, for instance, don't like UTF-8). There are no doubt other files or file types that could be added; this is just intended to provide an initial foundation.

**Review by TrueBrain — DISMISSED** [review]
You give a finger .. and we want to eat the hand :D

**Comment by TrueBrain** [comment]
Sorry, there was this "update branch" button I just had to click :D Please rebase your branch so you can merge it :)

**Comment by orudge** [comment]
I've removed your merge and rebased this onto the current OpenTTD master, is that what you meant? Also the CI checks have failed so the merge option is greyed out, unless you just want me to push directly to the repository?

**Pull request closed without merging**